policy-controller/k8s/status/src/index.rs

```diff
@@ -114,9 +131,12 @@ impl Controller {
                 biased;
                 res = self.claims.changed() => {
                     res.expect("Claims watch must not be dropped");
-                    tracing::debug!("Lease holder has changed");
                     let claim = self.claims.borrow_and_update();
+                    let was_leader = self.leader;
                     self.leader = claim.is_current_for(&self.name);
+                    if was_leader != self.leader {
+                        tracing::debug!(leader = %self.leader, "Leadership changed");
+                    }
                 }
                 // If this policy controller is not the leader, it should
                 // process through the updates queue but not actually patch
```

Comment:
I can't quite comment on the line I want, but this comment here does not accurately reflect what this code does, and that is a subtle bug lurking:

```rust
                Some(Update { id, patch}) = self.updates.recv(), if self.leader => {
```

This condition mans that we will only ever check the channel if we are currently the leader. This means that if there is a leadership change (and we lose leadership), the channel can end up holding patches indefinitely, and when leadership is resumed we'll process very old patches. This if condition must be moved within the body of the block, and we should drop updates while we're not the leader.